File: C_Sharp/p1071.cs
using System.Numerics;

namespace p1071{
    public class Solution {
        public string GcdOfStrings(string str1, string str2) {
            if(str1+str2 != str2 + str1){
                return "";
            }
            int resultLength = ((int)BigInteger.GreatestCommonDivisor(str1.Length, str2.Length));
            return str1.Substring(0,resultLength);
        }
    }
}

File: C_Plus_Plus/1001-2000/p1071.cpp
#include <string>
#include <numeric>
using namespace std;

class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        /**
         * @brief if str1 and str2 have a common divisor then the strings will contain the same pattern
         * this means the strings concatenated together should be the same no matter the order of
         * concatenation
         */
        if(str1+str2 != str2+str1){
            return "";
        }
        /**
         * @brief we know at this point that their is a common divisor. If we know there is a common divisor
         * we can then find the common divisors of their length and return a substring of that length
         */
        int resultLength = gcd(str1.length(), str2.length());
        return str1.substr(0, resultLength);
    }
};